components/organisms/SignUpForm/SignUpFormContent.js

```diff
@@ -67,55 +67,71 @@ export const LinksContainer = styled.div`
   }
 `;
 
-const SignUpFormContent = ({ onSubmit }) => {
-  const form = {
-    fields: [
-      {
-        type: 'text',
-        name: 'firstName',
-        title: 'First Name',
-        placeholder: 'First Name',
-        initialValue: '',
-        validationSchema: Yup.string().required('This field is required'),
-      },
-      {
-        type: 'text',
-        name: 'lastName',
-        title: 'Last Name',
-        placeholder: 'Last Name',
-        initialValue: '',
-        validationSchema: Yup.string().required('This field is required'),
-      },
-      {
-        type: 'email',
-        name: 'email',
-        title: 'Email',
-        placeholder: 'Email',
-        initialValue: '',
-        validationSchema: Yup.string()
-          .email('The email must be valid!!')
-          .required('This field is required'),
-      },
-      {
-        type: 'password',
-        name: 'password',
-        title: 'Password',
-        placeholder: '',
-      },
-      {
-        type: 'submit',
-        name: 'sign_up',
-        initialValue: 'Sign Up',
-      },
-    ],
-    submit: values => onSubmit(values),
+const SignUpFormContent = ({ onSubmit, activeForm }) => {
+  const isSignUp = activeForm === 'signUp';
+  const fieldNames = {
+    firstName: isSignUp,
+    lastName: isSignUp,
+    email: true,
+    password: true,
+    signUp: isSignUp,
+    signIn: !isSignUp,
   };
 
+  const allFields = [
+    {
+      type: 'text',
+      name: 'firstName',
+      title: 'First Name',
+      placeholder: 'First Name',
+      initialValue: '',
+      validationSchema: Yup.string().required('This field is required'),
+    },
+    {
+      type: 'text',
+      name: 'lastName',
+      title: 'Last Name',
+      placeholder: 'Last Name',
+      initialValue: '',
+      validationSchema: Yup.string().required('This field is required'),
+    },
+    {
+      type: 'email',
+      name: 'email',
+      title: 'Email',
+      placeholder: 'Email',
+      initialValue: '',
+      validationSchema: Yup.string()
+        .email('The email must be valid!!')
+        .required('This field is required'),
+    },
+    {
+      type: 'password',
+      name: 'password',
+      title: 'Password',
+      placeholder: '',
+      initialValue: '',
+    },
+    {
+      type: 'submit',
+      name: 'signUp',
+      initialValue: 'Sign Up',
+    },
+    {
+      type: 'submit',
+      name: 'signIn',
+      initialValue: 'Sign In',
+    },
+  ];
+  const currentFormFields = [];
```

Comment:
`const currentFormFields = {} `? and looks like submit property `submit: values => onSubmit(values)` should be here too